rplugin/python3/gdb/scm.py

```diff
@@ -7,6 +7,7 @@ class BaseScm(Common):
     '''Common SCM implementation for the integrated backends.'''
     def __init__(self, common, cursor, win):
         super().__init__(common)
+        self.vim = common.vim
```

Comment:
Isn't `self.vim` already inherited from `Common`?

Reply:
I should admit, SCM was a bad name. It meant to be FSM, finite state machine. And was quite sure there was a library in the boost called SCM, but I don't seem to be able to find it anymore.
Probably, I should fix the name of the class to avoid any future confusion.
I guess the best place to add another aspect could be the class `App`. It ties all the entities together, and anything needed is there already. Naturally, the behaviour of the `scm` could be tweaked there too with some kind of parameter.